Get rid of DefaultNumBailouts

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #74186
* #74012
* #73938
* #73876
* __->__ #73689
* #73875

The number of bailouts of each executor should just be instantiated from whatever the current fusion strategy is, which is then stored on the executor.

Commit:
Get rid of DefaultNumBailouts

[ghstack-poisoned]

Commit:
Update on "Get rid of DefaultNumBailouts"


The number of bailouts of each executor should just be instantiated from whatever the current fusion strategy is, which is then stored on the executor. 

[ghstack-poisoned]

Commit:
Update on "Get rid of DefaultNumBailouts"


The number of bailouts of each executor should just be instantiated from whatever the current fusion strategy is, which is then stored on the executor. 

[ghstack-poisoned]

Commit:
Update on "Get rid of DefaultNumBailouts"


The number of bailouts of each executor should just be instantiated from whatever the current fusion strategy is, which is then stored on the executor. 

[ghstack-poisoned]

Commit:
Update on "Get rid of DefaultNumBailouts"


The number of bailouts of each executor should just be instantiated from whatever the current fusion strategy is, which is then stored on the executor. 

[ghstack-poisoned]

Commit:
Update on "Get rid of DefaultNumBailouts"


The number of bailouts of each executor should just be instantiated from whatever the current fusion strategy is, which is then stored on the executor. 

[ghstack-poisoned]

Commit:
Update on "Get rid of DefaultNumBailouts"


The number of bailouts of each executor should just be instantiated from whatever the current fusion strategy is, which is then stored on the executor. 

[ghstack-poisoned]

eellison commented on test/test_jit.py:
```diff
@@ -203,11 +203,6 @@ def doAutodiffCheck(testname):
-# we haven't tested every single test in this file
-# but we enable FULL_PROFILER for a large subset
-# of the tests with "with enable_profiling_mode_for_profiling_tests"
-torch._C._jit_set_profiling_mode(False)
```
also updates this finally, needed to fix a couple tests @Krovatkin

Commit:
Update on "Get rid of DefaultNumBailouts"


The number of bailouts of each executor should just be instantiated from whatever the current fusion strategy is, which is then stored on the executor. 

[ghstack-poisoned]

davidberard98 reviewed (approved):
Awesome! Looks good to me

nit: is there a reason we can't get rid of all the bailOut arguments, e.g. in graph_executor.cpp? It doesn't seem like they are used anywhere

Commit:
Update on "Get rid of DefaultNumBailouts"


The number of bailouts of each executor should just be instantiated from whatever the current fusion strategy is, which is then stored on the executor. 

[ghstack-poisoned]

eellison commented:
> nit: is there a reason we can't get rid of all the bailOut arguments, e.g. in graph_executor.cpp? It doesn't seem like they are used anywhere

i think some of it is still being used for passing in the number of remaining specializations.. but yea some cleanup could probably still be done
